src/server/src/modules/job.ts

```diff
@@ -30,28 +30,40 @@ export const createJob = (
     maxSalary: IJob['maxSalary'],
     departmentId: IJob['departmentId']
 ) => {
-    if (expirationDate == null) {
+    const startDateAsDate = new Date(startDate);
+
+    let expirationDateAsDate: Date;
+    if (expirationDate) {
+        expirationDateAsDate = new Date(expirationDate);
+    } else {
         // Set expiration date to 2 years after the current date.
-        expirationDate = new Date();
-        expirationDate.setFullYear(expirationDate.getFullYear() + 2);
+        expirationDateAsDate = new Date();
+        expirationDateAsDate.setFullYear(expirationDateAsDate.getFullYear() + 2);
     }
-    if (maxSalary == null) {
+
+    let endDateAsDate;
+    if (endDate) {
+        endDateAsDate = new Date(endDate);
+    }
+
+    if (!maxSalary) {
         maxSalary = minSalary;
     }
+
     const repository = getRepository(Job);
     const jobToInsert = repository.create({
-        targetYears,
-        hoursPerWeek,
-        description,
-        expirationDate,
-        startDate,
-        type,
-        title,
-        status,
-        minSalary,
-        departmentId,
-        endDate,
-        maxSalary
+        targetYears: targetYears,
```

Comment:
I think what you before was fine. 
{
        targetYears,
        hoursPerWeek,
        ....
}

Reply:
For some reason, it didn't work.